src/ros_filter.cpp

```diff
@@ -90,6 +90,39 @@ namespace RobotLocalization
     topicSubs_.clear();
   }
 
+  template<typename T>
+  void RosFilter<T>::reset()
+  {
+    // Get rid of any initial poses (pretend we've never had a measurement)
+    initialMeasurements_.clear();
+    previousMeasurements_.clear();
+    previousMeasurementCovariances_.clear();
+
+    // Clear the measurement queue.
+    // This prevents us from immediately undoing our reset.
+    while (!measurementQueue_.empty() && ros::ok())
+    {
+      measurementQueue_.pop();
+    }
+
+    ros::getGlobalCallbackQueue()->clear();
```

Comment:
So I found out through some not-so-fun testing that this method does what it advertises, but if called from within a callback, it may cause the callback *itself* to exit prematurely, which means all the lines after this one may not get called. If possible, can we put this call at the very end of this method?